packages/@react-stately/color/src/Color.ts

```diff
@@ -80,8 +99,113 @@ abstract class Color implements IColor {
 
     return {xChannel: xCh, yChannel: yCh, zChannel: zCh};
   }
+
   abstract getColorChannels(): [ColorChannel, ColorChannel, ColorChannel]
+
+  getColorName(locale: string): string {
+    // Convert to oklch color space, which has perceptually uniform lightness across all hues.
+    let [l, c, h] = toOKLCH(this);
+
+    let strings = LocalizedStringDictionary.getGlobalDictionaryForPackage('@react-stately/color') || dictionary;
+    if (l > 0.999) {
+      return strings.getStringForLocale('white', locale);
+    }
+
+    if (l < 0.001) {
+      return strings.getStringForLocale('black', locale);
+    }
+
+    let hue: string;
+    [hue, l] = this.getOklchHue(l, c, h, locale);
```

Comment:
I see `l` change as a result of this calculation, but I don't quite understand why. I know it's for orange because it dips into brown very quickly, but wouldn't the `l` already be correct because it's one of the channels returned by the conversion to OKLCH?

if we just got this from the sample conversion code, that's fine, I might've missed it while reading through

Reply:
It comes from this line: https://github.com/adobe/react-spectrum/pull/6180/files/c05e757846a4745b9e80cb2ec9fbf54e595abb77#diff-2737484a486e2634802c43f60406fc1231698480673f4268bd1e8f029448bcccR185

Basically, if the hue is orange, we need to adjust the lightness so that "dark orange" comes before "brown", so we split the lightness range in half. Otherwise orange would always be considered light/very light.